core/domain/feature_gating_services.py

```diff
@@ -0,0 +1,191 @@
+        {
+            'server_mode': _get_server_mode()
+        }
+    )
+
+
+def get_all_feature_flag_dicts():
+    """Returns dict representations of all feature flags.
+
+    Returns:
+        dict. The keys are the feature names and the values are the dict
+        mappings of all fields of the feature flags.
+    """
+    return [
+        registry.Registry.get_platform_parameter(name).to_dict()
+        for name in ALL_FEATURES_LIST
+    ]
+
+
+def evaluate_all_feature_flag_value_for_client(context):
+    """Evaluates and returns the values for all feature flags.
+
+    Args:
+        context: EvaluationContext. The context used for evaluation.
+
+    Returns:
+        dict. The keys are the feature names and the values are boolean
+        results of corresponding flags.
+    """
+    return _evaluate_feature_flag_values_for_context(
+        ALL_FEATURES_LIST, context)
+
+
+def evaluate_feature_flag_value_for_server(feature_name):
+    """Evaluates and returns the values of the feature flag, using context
+    from the server only.
+
+    Args:
+        feature_name: str. The name of the feature flag that needs to
+            be evaluated.
+
+    Returns:
+        bool. The value of the feature flag, True if it's enabled.
+    """
+    context = _create_evaluation_context_for_server()
+    values_dict = _evaluate_feature_flag_values_for_context(
+        [feature_name], context)
+    return values_dict[feature_name]
+
+
+def update_feature_flag_rules(
+        feature_name, committer_id, commit_message, new_rule_dicts):
+    """Updates the feature flag's rules.
+
+    Args:
+        feature_name: str. The name of the feature to update.
+        committer_id: str. ID of the committer.
+        commit_message: str. The commit message.
+        new_rule_dicts: list(dist). A list of dict mappings of all fields
+            of PlatformParameterRule object.
+    """
+    if feature_name not in ALL_FEATURES_NAMES_SET:
+        raise Exception('Feature flag not exist: %s.' % feature_name)
+
+    registry.Registry.update_platform_parameter(
+        feature_name, committer_id, commit_message, new_rule_dicts)
+
+
+# TODO(#10211): Currently Oppia runs in either of the two modes:
+# dev or prod. There should be another mode 'test' added for QA testing,
+# once it is added, this function needs to be updated to take that into
+# consideration.
+def _get_server_mode():
+    """Returns the running mode of Oppia.
+
+    Returns:
+        str. The server mode of Oppia, 'dev' if Oppia is running in development
+        mode, 'prod' if in production mode.
+    """
+    if constants.DEV_MODE:
+        return param_domain.SERVER_MODES.dev
+    else:
+        return param_domain.SERVER_MODES.prod
+
+
+def _create_evaluation_context_for_server():
+    """Returns evaluation context with information of the server.
+
+    Returns:
+        EvaluationContext. The context for evaluation.
+    """
+    return param_domain.EvaluationContext.from_dict(
```

Comment:
Should we create a ` param_domain.EvaluationContext.create_default_evaluation_context(server_mode)` method and use it here? (I think that would make it simpler, but I'm not sure whether we always pass None for all the fields.)